packages/ti_stix/data_stream/indicator/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,371 @@
+          } else if (time_unit == 'h') {
+            ioc_expiration_date = updated_at.plusHours(Long.parseLong(time_value));
+          } else if (time_unit == 'm') {
+            ioc_expiration_date = updated_at.plusMinutes(Long.parseLong(time_value));
+          } else {
+            if (ctx.error == null) {
+              ctx.error = new HashMap();
+            }
+            if (ctx.error.message == null) {
+              ctx.error.message = new ArrayList();
+            }
+            ctx.error.message.add('invalid ioc_expiration_duration: using default 90 days');
+            ioc_expiration_date = updated_at.plusDays(90L);
+          }
+        } else {
+            ioc_expiration_date = updated_at.plusDays(90L);
+        }
+
+        ctx.stix.ioc_expiration_date = ioc_expiration_date;
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+
+  - script:
+      lang: painless
+      tag: script_set_ioc_expiration_reason
+      if: ctx.stix?.ioc_expiration_date != null
+      params:
+        valid_until: "Expiration set from valid_until field"
+        revoked: "Expiration set from revoked field"
+        default: "Expiration set by Elastic from the integration's parameter `IOC Expiration Duration`"
+      source: >
+        if (ctx.stix.valid_until != null) {
+          ctx.stix.ioc_expiration_reason = params.valid_until;
+        } else if (ctx.stix.revoked != null && ctx.stix.revoked == true) {
+          ctx.stix.ioc_expiration_reason = params.revoked;
+        } else {
+          ctx.stix.ioc_expiration_reason = params.default;
+        }
+
+  - date:
+      field: stix.ioc_expiration_date
+      tag: date_ioc_expiration_date
+      target_field: stix.ioc_expiration_date
+      formats:
+        - ISO8601
+        - UNIX
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+        - fail:
+            description: 'If this date processor fails, it leads to transform errors, hence failing the document for investigation.'
+            message: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed'
+  - rename:
+      field: _conf.ioc_expiration_duration
+      tag: rename_conf_ioc_expiration_duration
+      target_field: stix.ioc_expiration_duration
+      ignore_missing: true
+
+  - set:
+      field: threat.indicator.type
+      value: autonomous-system
+      if: ctx.stix?.pattern != null && ctx.stix.pattern.startsWith('[autonomous-system')
```

```suggestion
      if: ctx.stix?.pattern?.startsWith('[autonomous-system') == true
```
?

(with caveat about stix nullness)